platforms/core-configuration/file-operations/src/main/java/org/gradle/api/internal/file/DefaultProjectLayout.java

```diff
@@ -123,9 +120,9 @@ public FileCollection files(Object... paths) {
         return fileCollectionFactory.resolving(paths);
     }
 
-    /**
-     * A temporary home. Should be on the public API somewhere
-     */
+    // TODO: this "unused" method should be removed
+    // However, currently, expressions like 'layout.buildDirectory = file(...)' resolve to this method,
+    // and removing it makes those expressions fail the build
```

Comment:
I think if `DefaultProjectLayout` is created through our decorator (instead of just `new`ing it), I think we would create this method as a decoration... but does this also allow `layout.buildDirectory = "foo"`?

Reply:
>but does this also allow layout.buildDirectory = "foo"?

Yes, unfortunately. We even have some integration tests using that. It's too easy to imagine such usage is already spread in the ecosystem. So, I wonder, if we should be careful and issue a deprecation from this method and only remove it in 9.0